implementations/rust/ockam/ockam_api/src/nodes/models/show.rs

```diff
@@ -0,0 +1,76 @@
+use ockam_multiaddr::{
+    proto::{DnsAddr, Node, Tcp},
+    MultiAddr,
+};
+use serde::Serialize;
+
+use super::{
+    portal::{InletList, OutletList},
+    secure_channel::SecureChannelListenersList,
+    services::ServiceList,
+    transport::TransportList,
+};
+
+#[derive(Debug, Serialize)]
+pub struct ShowNodeResponse<'a> {
+    pub is_default: bool,
+    pub name: &'a str,
+    pub is_up: bool,
+    pub route: RouteToNode,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub identity: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none", flatten)]
+    pub tcp_listeners: Option<TransportList>,
```

Comment:
Would it work if you add the `rename` attribute here insteaf of in the `TransportList`? If you can do that, you would be able to leave the `ockam_api` structs unmodified, which would be desirable.

Reply:
That was my first attempt but flatten doesn't work with rename. It makes as the struct being flattened could hold multiple attributes instead of just one like we have. A possible workaround to leave the API structs unmodified would be using a custom serializer or taking out the `Vec` with the list instead of passing the whole `TransportList`, but I would say that in this particular cases specifying the name of the lists to be more meaningful like "transports" or "services" vs. just "list" is probably good.